src/renderers/shared/fiber/ReactFiberContext.js

```diff
@@ -57,6 +57,18 @@ function getUnmaskedContext(workInProgress : Fiber) : Object {
 }
 exports.getUnmaskedContext = getUnmaskedContext;
 
+function cacheContext(workInProgress : Fiber, unmaskedContext : Object, maskedContext : Object) {
+  if (isContextConsumer(workInProgress)) {
```

Comment:
>I assume that getMaskedContext returns emptyObject instead of null for a reason.

Because we want to pass an empty object to components. But we could do `|| emptyObject` this [here](https://github.com/bvaughn/react/blob/a06bb126abc75232b37e8231fb1fe48f2bffed09/src/renderers/shared/fiber/ReactFiberClassComponent.js#L211) (and similar places) instead.

>This all inside of ReactFiberContext. What's the other file?

I am talking about [these lines](https://github.com/bvaughn/react/blob/a06bb126abc75232b37e8231fb1fe48f2bffed09/src/renderers/shared/fiber/ReactFiberClassComponent.js#L210-L217) in `ClassComponent` specifically:

```js
// This will check if it's a context consumer:
const context = getMaskedContext(workInProgress, unmaskedContext);
// ...
// This will do the same check:
cacheContext(workInProgress, unmaskedContext, context);
```

I was proposing to do something like
```js
// Check once:
const needsContext = isContextConsumer(workInProgress);

const context = needsContext ? getMaskedContext(workInProgress, unmaskedContext) : emptyObject;
// ...
if (needsContext) {
  cacheContext(workInProgress, unmaskedContext, context);
}
```